src/lib/EventListener/PerformAccessCheckSubscriber.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\User\EventListener;
+
+use Ibexa\Contracts\User\Controller\AccessCheckController;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+
+final class PerformAccessCheckSubscriber implements EventSubscriberInterface
+{
+    public function onControllerArgumentsEvent(ControllerArgumentsEvent $event): void
+    {
+        $controller = $event->getController();
+        if (is_array($controller) && $controller[0] instanceof AccessCheckController) {
```

Comment:
I'm pretty sure this can also be an invoke'able object (with `__invoke` method). While we ourselves never use controllers like that, there is nothing stoping someone else from doing so - since it's in `Contracts` namespace.

Reply:
I was just moving this from admin-ui, but you are right - fixed.
Also, string callables with static methods are also possible so I tried to handle those as well.